File: eco-core/core-plugin/src/main/java/com/willfp/ecomc/Hardinator.java
package com.willfp.ecomc;

import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.PluginDependent;
import com.willfp.eco.util.NumberUtils;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Mob;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.event.entity.EntitySpawnEvent;
import org.jetbrains.annotations.NotNull;

public class Hardinator extends PluginDependent<EcoPlugin> implements Listener {
    public Hardinator(@NotNull EcoPlugin plugin) {
        super(plugin);
    }

    @EventHandler
    public void handle(@NotNull final EntityDamageByEntityEvent event) {
        if (event.getDamager() instanceof Mob) {
            event.setDamage(event.getDamage() * this.getPlugin().getConfigYml().getDouble("incoming-damage-multiplier"));
        }
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecomc/Hardinator.kt
package com.willfp.ecomc

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.util.tryAsPlayer
import org.bukkit.entity.Mob
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent

class Hardinator(private val plugin: EcoPlugin) : Listener {
    @EventHandler
    fun takeMore(event: EntityDamageByEntityEvent) {
        if (event.damager is Mob) {
            event.damage = event.damage * plugin.configYml.getDouble("incoming-damage-multiplier")
        }
    }

    @EventHandler
    fun dealLess(event: EntityDamageByEntityEvent) {
        val player = event.damager.tryAsPlayer() ?: return
        event.damage = event.damage * plugin.configYml.getDouble("outgoing-damage-multiplier")
    }
}
